lib/pbench/test/unit/common/test_config.py

```diff
@@ -1,68 +1,252 @@
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "hostport",
+                        "logger_host": "logger.example.com",
+                        "logger_port": "42",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.logger_type == "hostport"
+        assert config.logger_host == "logger.example.com"
+        assert config.logger_port == "42"
+
+    def test_no_logger_dir_option(self, monkeypatch):
+        """test no logger_dir option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "file",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.log_dir is None
+
+    def test_logger_dir(self, monkeypatch):
+        """verify logger_dir option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {"logger_type": "file", "log_dir": "/mock/log_dir"}
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.log_using_caller_directory is False
+        assert config.log_dir == "/mock/log_dir"
+
+    def test_no_logging_level_option(self, monkeypatch):
+        """test no logging_level option in config file"""
+
+        class mock_MockedConfigParser(MockedConfigParser):
+            def __init__(self):
+                self._sections = {
+                    "logging": {
+                        "logger_type": "file",
+                    }
+                }
+
+        monkeypatch.setattr(configtools, "file_list", self.mock_file_list)
+        monkeypatch.setattr(ConfigParser, "__init__", mock_MockedConfigParser.__init__)
+        monkeypatch.setattr(ConfigParser, "get", MockedConfigParser.get)
+        config = PbenchConfig("pbench.cfg")
+        assert config.default_logging_level == "INFO"
```

Comment:
This test looks to me to be identical to `test_logger()` except for the assertion at line 201. I recommend that you move the assertion from here to that test and omit this test.

Reply:
yeah